hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/HddsDispatcher.java

```diff
@@ -304,10 +313,25 @@ private ContainerCommandResponseProto dispatchRequest(
       audit(action, eventType, params, AuditEventStatus.FAILURE, ex);
       return ContainerUtils.logAndReturnError(LOG, ex, msg);
     }
+    long handlingStartTime = Time.monotonicNow();
     responseProto = handler.handle(msg, container, dispatcherContext);
     if (responseProto != null) {
+      long requestDurationMs = Time.monotonicNow() - startTime;
+      long handlingDurationMs = Time.monotonicNow() - handlingStartTime;
```

Comment:
We should add a metric for this that can be pushed to prometheus.

Reply:
+1. 
If we expose those timings as metrics with proper quantile, maybe only  `avg` and `p99`, we may not need warning logs at all. p99 would report the outliners which we can proactively look at or create alerts on.

Reply:
There's already metric for the handler.
The warning is for something is really really off and shouldn't flood the log file.
My experience is that most people don't set up alerts (because they didn't even know where to look at), and it becomes hard to look at in a retrospective. Having a log warning when things is really bad allows quick troubleshooting.

Reply:
That makes sense. A warning log sounds easier for operators to find out as all of would scan the application logs.